File: app/src/main/java/fi/masaz/celatum/ItemAdapter.java
package fi.masaz.celatum;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import fi.masaz.celatum.room.Item;

public class ItemAdapter extends RecyclerView.Adapter<ItemAdapter.ItemViewHolder> {
    private final List<Item> itemList;
    private final OnItemClickListener onItemClickListener;

    public ItemAdapter(List<Item> itemList, OnItemClickListener listener) {
        this.itemList = itemList;
        this.onItemClickListener = listener;
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder holder, int position) {
        Item item = itemList.get(position);
        holder.mItemTitle.setText(item.title);
        holder.mImageView.setImageResource(Tools.getIcon(item.icon));

        holder.itemView.setOnClickListener(v -> {
            if (onItemClickListener != null) {
                onItemClickListener.onItemClick(item);
            }
        });
    }

    @Override
    public int getItemCount() {
        return itemList.size();
    }

    public class ItemViewHolder extends RecyclerView.ViewHolder {
        TextView mItemTitle;
        ImageView mImageView;

        public ItemViewHolder(View itemView) {
            super(itemView);
            mItemTitle = itemView.findViewById(R.id.item_title);
            mImageView = itemView.findViewById(R.id.item_icon);
        }
    }
}

File: app/src/main/java/fi/masaz/celatum/ItemAdapter.kt
package fi.masaz.celatum

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import fi.masaz.celatum.ItemAdapter.ItemViewHolder
import fi.masaz.celatum.room.Item

class ItemAdapter(
    private var itemList: List<Item>,
    private val onItemClickListener: OnItemClickListener?
) : RecyclerView.Adapter<ItemViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_layout, parent, false)
        return ItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        val item = itemList[position]
        holder.mItemTitle.text = item.title
        holder.mImageView.setImageResource(Tools.getIcon(item.icon))

        holder.itemView.setOnClickListener { v: View? ->
            onItemClickListener?.onItemClick(item)
        }
    }

    override fun getItemCount(): Int {
        return itemList.size
    }

    fun setItemList(newItemList: List<Item>) {
        this.itemList = newItemList
        notifyDataSetChanged() // Notify the adapter that the data has changed
    }

    inner class ItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mItemTitle: TextView = itemView.findViewById(R.id.item_title)
        var mImageView: ImageView = itemView.findViewById(R.id.item_icon)
    }
}
